src/main/java/com/example/todo_restapi/service/TaskService.java

```diff
@@ -18,16 +19,20 @@ public class TaskService {
 
 
     public Long saveTask(TaskDto taskDto) {
-        Tasks tasks = new Tasks(taskDto.getTitle(), taskDto.getDescription(),
+        Tasks task = new Tasks(taskDto.getTitle(), taskDto.getDescription(),
                 taskDto.getPriority(), Instant.now(), false);
-        taskRepository.save(tasks);
-        return tasks.getId();
+        Tasks savedTask = taskRepository.save(task);
+        return savedTask.getId();
     }
 
     public Optional<TaskDto> getTaskById(Long id) {
-        return taskRepository.findById(id)
-                .map(task -> new TaskDto(task.getId(), task.getTitle(), task.getDescription(),
-                        task.getPriority(), task.getCreateDate(), task.isCompleted()));
+        if (id > 0 && id < taskRepository.count()) {
```

Comment:
Nie rozumiem, po co to sprawdzenie. Spróbuj pobrać, a jak się nei uda to rzuć excpetion